File: examples/solution/Kernel/Example.Kernel/Money.cs
namespace Example.Kernel;

/// <summary>
/// An amount of money. A value, not an entity: two amounts of the same
/// size in the same currency are the same amount.
/// </summary>
public readonly record struct Money(decimal Amount, string Currency)
{
    /// <summary>
    /// Adds two amounts. Mixing currencies is refused rather than guessed at.
    /// </summary>
    /// <param name="other">The amount to add.</param>
    /// <returns>The sum, in the same currency.</returns>
    public Money Add(Money other)
    {
        if (!string.Equals(Currency, other.Currency, StringComparison.Ordinal))
        {
            throw new InvalidOperationException(
                $"Cannot add {other.Currency} to {Currency}: there is no rate here to convert with.");
        }

        return this with { Amount = Amount + other.Amount };
    }

    /// <summary>Renders the amount for display.</summary>
    public override string ToString() => $"{Amount:0.00} {Currency}";
}

File: examples/php/src/Kernel/Money.php
<?php
declare(strict_types=1);

namespace Example\Kernel;

/** An amount of money: a value, not an entity. */
final class Money
{
    public function __construct(
        private readonly float $amount,
        private readonly string $currency,
    ) {
    }

    public function times(float $factor): self
    {
        return new self($this->amount * $factor, $this->currency);
    }
}
